### PR TITLE
Update locale vi

### PR DESCRIPTION
**Pull request by lihop**
@trongthanh Please feel free to add comments here or branch of a create your own changes.

### CODE REVIEW

**Comment by lihop** [comment]
@trongthanh When I created this fork I just copy & pasted over the new en-US locale and then filled in the values from the CLDR data. I think I need to re-integrate your version which is slightly different from the [CLDR data](https://www.unicode.org/cldr/charts/32/summary/vi.html).

**Comment by trongthanh** [comment]
@lihop,

Let me take a closer look this evening. FWIW, CLDR data may not represent current usage of the speaker of the language, as you may notice the native speakers of Chinese and Thai had commented some errors in another PR. Ultimately, we want the translation sounds naturally. 

I'll consult instances from local online newspaper and provide proof so that we both agree on the correction (if any).

**Comment by trongthanh** [comment]
In favor of "X tiếng trước" instead of "X giờ trước":
screenshot at 2018-08-08 15-30-04

**Comment by trongthanh** [comment]
OK. I'll take previous comment back. It seems more news outlets use "giờ" instead of "tiếng".

**Comment by lihop** [comment]
@trongthanh Ok. I've finished updating everything. I've tried to keep everything as close as possible to the previous version.

File: src/locale/vi/_lib/localize/index.js [hunk 1]
```diff
@@ -1,53 +1,113 @@
...
+  wide: ['Quý 1', 'Quý 2', 'Quý 3', 'Quý 4']
 }
 
+// Note: in English, the names of days of the week and months are capitalized.
+// If you are making a new locale based on this one, check if the same is true for the language you're working on.
+// Generally, formatted dates should look like they are in the middle of a sentence,
+// e.g. in Spanish language the weekdays and months should be in the lowercase.
 var monthValues = {
+  narrow: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'],
+  short: ['Thg 1', 'Thg 2', 'Thg 3', 'Thg 4', 'Thg 5', 'Thg 6', 'Thg 7', 'Thg 8', 'Thg 9', 'Thg 10', 'Thg 11', 'Thg 12'],
```
Comment (trongthanh) [inline src/locale/vi/_lib/localize/index.js]: I don't think `monthValues` has **short** form as per i18n documentation and the format notation for Months are: M (number), Mo (ordinal number), MM (narrow), MMM (abbreviated), MMMM (wide).

**Comment by trongthanh** [comment]
I just pushed some changes with comments inside and fix the tests.

Although all tests pass now, but I think we should add some more for cases that only applicable for Vietnamese, so that it'll thoroughly validate the updates we make for VI locale.

**Comment by trongthanh** [comment]
I just changed relative distance hours from "tiếng" back to "giờ". Reasons:

- Most news outlets use "giờ". I can only find instances of "tiếng" from Thanhnien.vn so far.
- According to [tratu.soha.vn](http://tratu.soha.vn/dict/vn_vn/Ti%E1%BA%BFng), "tiếng" is used in spoken informal context.
- Writing "giờ" is obviously shorter and more space-saving than "tiếng".

File: src/locale/vi/_lib/localize/index.js [hunk 2]
```diff
@@ -1,53 +1,113 @@
...
   // format being localized with abbreviations found in some systems (SÁng / CHiều);
   // however, personally, I don't think `Chiều` sounds appropriate for `PM`
-  // TODO: for `long` version, the values should be 'sáng' 'trưa' 'chiều' 'tối'
-  uppercase: ['AM', 'PM'],
-  lowercase: ['am', 'pm'],
-  long: ['SA', 'CH']
+  narrow: {
+    am: 'a',
+    pm: 'p',
+    midnight: 'nửa đêm',
```
Comment (songoten28) [inline src/locale/vi/_lib/localize/index.js]: In vietnamese : 
"nửa đêm", not "nưả đêm"

File: src/locale/vi/_lib/localize/index.js [hunk 3]
```diff
@@ -1,53 +1,113 @@
...
+    night: 'đêm'
+  },
+  abbreviated: {
+    am: 'AM',
+    pm: 'PM',
+    midnight: 'nửa đêm',
+    noon: 'TR',
+    morning: 'sáng',
+    afternoon: 'chiều',
+    evening: 'tối',
```
Comment (songoten28) [inline src/locale/vi/_lib/localize/index.js]: and "tối" not "tôí"

File: src/locale/vi/_lib/localize/index.js (hunk 2 above)
Comment (trongthanh) [reply src/locale/vi/_lib/localize/index.js]: I think you pulled an old version. There's no "nưả đêm" as mentioned.

File: src/locale/vi/_lib/localize/index.js (hunk 3 above)
Comment (trongthanh) [reply src/locale/vi/_lib/localize/index.js]: This weird. There's no occurrence of "tôí". Are you viewing with full Vietnamese-supported font?

Comment (songoten28) [reply src/locale/vi/_lib/localize/index.js]: oh sorry, my pc was miss some font of vietnamese

**Comment by trongthanh** [comment]
I think these updates are good to be merged into main project for now. If there are updates, let's do in another PR.

**Comment by lihop** [comment]
Ok. I will open a pull request to date-fns.

**Comment by lihop** [comment]
Opened pr to upstream in https://github.com/date-fns/date-fns/pull/846

**Pull request closed without merging**